### PR TITLE
Pass up datagrams that are routed to this device but have no further route.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

**Comment by darconeous** [comment]
Isn't this going to put all border router traffic through what is nominally an error path? (Looking at that `otDumpDebugIp6()` line)

**Comment by jwhui** [comment]
@darconeous, I've updated the PR.  Let me know if that better separates out the error path.

File: src/core/thread/thread_netif.cpp [hunk 1]
```diff
@@ -129,7 +129,18 @@ ThreadError ThreadNetif::GetLinkAddress(Ip6::LinkAddress &address) const
...
-    return mNetworkDataLeader.RouteLookup(source, destination, prefixMatch, NULL);
+    ThreadError error;
+    uint16_t rloc;
+
+    SuccessOrExit(error = mNetworkDataLeader.RouteLookup(source, destination, prefixMatch, &rloc));
+
+    if (rloc == mMleRouter.GetRloc16())
+    {
+        error = kThreadError_NoRoute;
+    }
```
Comment (abtink) [inline src/core/thread/thread_netif.cpp]: Do I get this correct? 

We return an error if the  `rloc` of the dst is same as ours (this is I guess the case where the msg should be passed up to host, right?), and then we pass up everything that we have no route for to host (in the `else` part in the `ForwardMessage()`).

Comment (jwhui) [inline src/core/thread/thread_netif.cpp]: That's the intent.

Comment (abtink) [inline src/core/thread/thread_netif.cpp]: Thanks.

**Pull request merged**